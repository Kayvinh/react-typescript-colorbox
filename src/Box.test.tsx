import React from "react";
import Box from "./Box";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';

const testBox = <Box
    id="1"
    width="10"
    height="5"
    backgroundColor="green"
    remove={function (id: string) { return id }}
/>

describe("Box", function () {

    test("renders without crashing", function () {
        render(testBox)
    })

    test("has the proper style", function () {
        const { container } = render(testBox);
        const box = container.querySelector(".Box-box")
        expect(box).toBeInTheDocument();
        expect(box).toHaveStyle({
            'background-color': 'green',
            height: '5em',
            width: '10em',
        })
    })

    test("matches snapshot", function() {
        const {container} = render(testBox);

        expect(container).toMatchSnapshot();
    })
})