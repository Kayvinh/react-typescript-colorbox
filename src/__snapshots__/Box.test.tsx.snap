// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div>
  <div
    class="Box"
  >
    <div
      class="Box-box"
      style="height: 5em; width: 10em; background-color: green;"
    />
    <button
      class="Box-removeBtn"
    >
      Remove The Box!
    </button>
  </div>
</div>
`;
